### PR TITLE
Deprecate mlab.{apply_window,stride_repeat}.

### PR DESCRIPTION
**Pull request by anntzer**
Altogether, they take more than 100 lines of implementation and many
hundred lines of tests, for what amounts to an array multiplication
optionally preceded by calling a user-supplied callable (see how
apply_window is inlined in _spectral_helper; stride_repeat is strictly a
helper for apply_window).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by dstansby** [comment]
Looks good to me. It looks like codecov isn't working for any of your PRs, is there any chance you could take a look into why?

**Comment by anntzer** [comment]
Actually it's at https://codecov.io/gh/matplotlib/matplotlib/commit/b57c39af1ff295d29cf80bdd227b238106a8b6e8, though I don't know why github did not link it.

**Comment by dstansby** [comment]
Odd, I wonder why it's not working just for you?

File: lib/matplotlib/mlab.py [hunk 1]
```diff
@@ -520,29 +522,34 @@ def _spectral_helper(x, y=None, NFFT=None, Fs=None, detrend_func=None,
             numFreqs = pad_to//2 + 1
         scaling_factor = 2.
 
+    if not np.iterable(window):
```
Comment (tacaswell) [inline lib/matplotlib/mlab.py]: would `if callbable(window)` make more sense?

Comment (anntzer) [reply lib/matplotlib/mlab.py]: The logic is the same in apply_window, I don't think handling the behavior change is worth it...

**Pull request merged**